app/pages/project/index.jsx

```diff
@@ -159,18 +158,9 @@ class ProjectPageController extends React.Component {
             }));
         })
     :
-      Promise.resolve(apiClient.type('project_preferences').create({
-        id: 'GUEST_PREFERENCES_DO_NOT_SAVE',
-        links: {
-          project: project.id
-        },
-        preferences: {}}));
+      Promise.resolve(null);
```

Comment:
I'm a bit worried this might cause some conflicts elsewhere in code. Beforehand, even non-logged-in users were given a `preferences` object. Curious if existence checks will need to be made elsewhere in code.

Reply:
Possibly here? https://github.com/zooniverse/Panoptes-Front-End/blob/e99fa0f4df6b4270f7d984e985656653431426ab/app/pages/project/classify.jsx#L192

Reply:
Or here? https://github.com/zooniverse/Panoptes-Front-End/blob/fe0c87a8ef589ef6c662bbb350ed35f8b1b8e895/app/classifier/drawing-tools/grid.cjsx#L105

Reply:
https://github.com/zooniverse/Panoptes-Front-End/blob/fe0c87a8ef589ef6c662bbb350ed35f8b1b8e895/app/classifier/tasks/drawing/marking-initializer.cjsx#L85

Reply:
I think this is worth checking with @srallen but my understanding is that the fake user preferences added here were just there to persist the selected workflow.

Reply:
There are likely other side effects, possibly also for tracking whether a user has seen the tutorial too. Removing this might cause the tutorial to pop up every time an anonymous user loads the classify page during a session.

Reply:
Will is right that some existential checks will need to be added to certain lines of code too.

Reply:
The tutorial here only loads once for me when I'm not logged in. https://pr-5035.pfe-preview.zooniverse.org/projects/eatyourgreens/-project-testing-ground/classify

Now that we've got the redux store, we should try not to use the API client for storage.

Reply:
Drawing tools with details tasks are broken on that project though. 😞

Reply:
My preferred solution would be not to rely on the existence of user preferences for anything that can be done by an anonymous user, so set a sensible default here or set sensible defaults in components that take a preferences prop.

Reply:
Tutorials use a global variable to store session data. That could be handled by the store too, if only so we know where to look for it.
`window.tutorialsCompletedThisSession = completedThisSession;`

Reply:
Ah yes, I agree that is the preferred solution and the route we'll take with the new classifier.

Reply:
The marking initialiser should be fixed now. I'm not sure if the grid tool was ever used in a project. Can that code can go (but maybe in a separate PR.)

Reply:
I've checked Gravity Spy on this branch. The workflow assignment popup only shows up if I'm logged in. 'Take me to the next level' updates my preferences, but doesn't refresh the workflow so I guess that needs to call `actions.classifier.setWorkflow` too. I probably need to add a new action that takes a workflow ID and fetches that workflow from Panoptes.

Reply:
Ok I've added a quick, and hacky (sorry!), fix for workflow assignment that requests the new workflow from Panoptes, then updates the store.

I haven't touched the grid tool because I'm not sure what the best approach is there. It seems like,the way it's written at the moment, that tool is intended for logged-in users anyway since the mark annotation is using user preferences (if I'm reading the code correctly.)

Reply:
I believe the grid tool can be deleted in a separate PR. I'll open an issue about it now and bring it up in a channel.

Reply:
If it's still needed, I'd suggest rewriting it to store data using redux.

Reply:
The grid tool appears to work in the dev classifier even if I'm not logged in. Not logged in, I can create a grid, give it a name and use the saved grid on subsequent classifications in the dev classifier.

Not logged in, I see API errors because creating a new grid tries to save user preferences, so I'm guessing that grids are saved so that they don't have to be created again on the next visit to the site. However, save won't work with the fake preferences that used to be created here, so I don't think this PR changes the grid tool save feature for anonymous users.

Ideally, the grid tool should check whether there's a user before attempting to save to Panoptes.

Reply:
Confirmed on https://master.pfe-preview.zooniverse.org/dev/classifier that the preferences save error exists on master, and isn't introduced by this change.